src/woo_publications/publications/tests/test_topic_api.py

```diff
@@ -0,0 +1,165 @@
+from django.utils.translation import gettext as _
+
+from freezegun import freeze_time
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from woo_publications.accounts.tests.factories import UserFactory
+from woo_publications.api.tests.mixins import (
+    APIKeyUnAuthorizedMixin,
+    APITestCaseMixin,
+    TokenAuthMixin,
+)
+
+from ..constants import PublicationStatusOptions, TopicStatusOptions
+from .factories import PublicationFactory, TopicFactory
+
+AUDIT_HEADERS = {
+    "AUDIT_USER_REPRESENTATION": "username",
+    "AUDIT_USER_ID": "id",
+    "AUDIT_REMARKS": "remark",
+}
+
+
+class TopicApiAuthorizationAndPermissionTests(APIKeyUnAuthorizedMixin, APITestCase):
+    def test_403_when_audit_headers_are_missing(self):
+        user = UserFactory.create()
+        self.client.force_authenticate(user=user)
+        list_endpoint = reverse("api:topic-list")
+        detail_endpoint = reverse("api:topic-detail", kwargs={"uuid": str(uuid4())})
+
+        with self.subTest(action="list"):
+            response = self.client.get(list_endpoint, headers={})
+
+            self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        with self.subTest(action="retrieve"):
+            response = self.client.get(detail_endpoint, headers={})
+
+            self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_api_key_result_in_401_with_wrong_credentials(self):
+        publication = PublicationFactory.create()
+        list_url = reverse("api:topic-list")
+        detail_url = reverse(
+            "api:topic-detail",
+            kwargs={"uuid": str(publication.uuid)},
+        )
+
+        # read
+        self.assertWrongApiKeyProhibitsGetEndpointAccess(list_url)
+        self.assertWrongApiKeyProhibitsGetEndpointAccess(detail_url)
+
+
+class PublicationApiTestsCase(TokenAuthMixin, APITestCaseMixin, APITestCase):
+    def test_list_topics(self):
+        with freeze_time("2024-09-25T12:00:00-00:00"):
+            topic = TopicFactory.create(
+                officiele_titel="title one",
+                omschrijving="bla bla bla",
+                publicatiestatus=TopicStatusOptions.published,
+                promoot=True,
+            )
+        with freeze_time("2024-09-24T12:00:00-00:00"):
+            topic2 = TopicFactory.create(
+                officiele_titel="title two",
+                omschrijving="description",
+                publicatiestatus=TopicStatusOptions.revoked,
+                promoot=False,
+            )
+
+        publication = PublicationFactory.create(onderwerpen=[topic])
+        publication2 = PublicationFactory.create(onderwerpen=[topic2])
+
+        response = self.client.get(reverse("api:topic-list"), headers=AUDIT_HEADERS)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        data = response.json()
+        self.assertEqual(data["count"], 2)
+
+        with self.subTest("first_item_in_response_with_expected_data"):
```

Comment:
can you please just write normal sentences instead of the `something_with_underscores`? The args and kwargs for subTest are displayed on test failure to help locate the failure, and otherwise it's a readability thing.